plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -139,6 +140,8 @@
     // SqlServer supports 2100 parameters in prepared statement, let's create a space for about 4 big IN predicates
     public static final int SQL_SERVER_MAX_LIST_EXPRESSIONS = 500;
 
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
```

Comment:
Will this pattern work for dates far into the future with 5 digit years? IIRC it should be `y..y-MM-dd`.

Reply:
Good point. Let's have a test with 5- and 6- digit dates. a pattern `y-MM-dd` may work. Or more correctly `u` instead of `y`.

Reply:
SQL Server does not support > 4 digit years.
https://docs.microsoft.com/en-us/sql/t-sql/data-types/date-transact-sql?view=sql-server-ver15

I agree we should have non 4 digit years tests for other connectors, I can do this along with support for dates < 1583 as a follow-up starting from PostgreSQL.

Reply:
> SQL Server does not support > 4 digit years.

Can you please add a test that would fail when they add support for that?
I don't think we're going to check the docs regularly with this in mind.

Reply:
Also, note that 5-digit years are not totally unexpected in practise.  Some people use `9999-12-31 23:59:59` as a "max value placeholder" (regardless whether this is actually a max value), and then such values can shift to `10000-01-01` when some timezone adjustements are applied.

Reply:
> Can you please add a test that would fail when they add support for that?
Added.